hr_recruitment_partner/init_hooks.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+#    Copyright (C) 2015 Daniel Reis
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp import SUPERUSER_ID
+from openerp.api import Environment
+
+
+def pre_init_hook(cr):
+    env = Environment(cr, SUPERUSER_ID, {})
+    env.cr.execute(
+        'UPDATE hr_applicant SET partner_id=1 WHERE partner_id IS NULL')
+
+
+def post_init_hook(cr, pool):
+    env = Environment(cr, SUPERUSER_ID, {})
+    # create a new partner for all applicants with no real partner assigned
+    applicants = env['hr.applicant'].with_context(active_test=False).\
+        search([('partner_id', '=', 1)], order='id')
```

Comment:
It's redundant now, but it's there just in case some other module changes the model's order.